### PR TITLE
LIDVID Query Wildcard Support

### LINKED ISSUE #177: As an API user I want to filter on lidvids with wildcards
See ticket https://github.com/NASA-PDS/pds-doi-ui/issues/25 which will use this feature.

### PR DESCRIPTION
**Pull request by collinss-jpl**
**Summary**
This PR updates the DOI service to allow users to provide Unix-style wildcards to LID's or LIDVID's when querying the local transaction database, either via the API or using the pds-doi-cmd CLI.

There has also been some cleanup of logging statements for the affected modules.

**Test Data and/or Report**
A unit test for querying LID/LIDVID using wildcards has been added.
[test.txt](https://github.com/NASA-PDS/pds-doi-service/files/6251195/test.txt)

**Related Issues**
- resolves #177

### CODE REVIEW

File: pds_doi_service/core/db/doi_database.py [hunk 1]
```diff
@@ -456,17 +458,86 @@ def _get_query_criteria_title(v):
     def _get_query_criteria_doi(v):
         return DOIDataBase._get_simple_in_criteria(v, 'doi')
 
+    @staticmethod
+    def _form_query_with_wildcards(column_name, search_tokens):
```
Comment (nutjob4life) [inline pds_doi_service/core/db/doi_database.py:462]: This is pretty nice partitioning and setup of the condition clause

Is there any chance that the `search_tokens` will contain dangerous strings like `; drop table *`?

**Review by nutjob4life — APPROVED** [review]
See the caveat on possible SQL injections to the `_form_query_with_wildcards` functions otherwise I say +1 to merge

**Comment by nutjob4life** [comment]
My vote is to 🛳 it, but I'll let @tloubrieu-jpl give final approval on that.

File: pds_doi_service/core/db/doi_database.py (hunk 1 above)
Comment (collinss-jpl) [reply pds_doi_service/core/db/doi_database.py:462]: The code itself is certainly not doing any *explicit* sanitation for malicious input, but my (rudimentary) understanding was that the usage of "named parameters" (placeholders in the query string and providing a mapping dictionary to `cursor.execute`) was supposed to protect against most(?) injection attacks. However, If a more defensive technique is required, I'm happy to add it.

**Review by tloubrieu-jpl — APPROVED** [review]
That works !

**Pull request merged**